### PR TITLE
Add wait for pending update method

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #6 

Additions: 
- waitOptions structure to communicate interval and timeOut to waitForPendingUpdate
- New error MeiliSearch.Error.TimeOut in case timeOut is reached
-

### CODE REVIEW

**Review by alallema — DISMISSED** [review]
LGTM! :tada:

File: Sources/MeiliSearch/Client.swift [hunk 1]
```diff
@@ -373,6 +373,27 @@ public struct MeiliSearch {
...
+
+    Providing an update id, returned by asynchronous MeiliSearch options, call are made
+    to MeiliSearch to check if the update has been processed or if it has failed.
+
+    - parameter UID:                 The unique identifier of the `Index`.
+    - parameter updateId:            The id of the update.
+    - parameter: options             Optionnal configuration for timeout and interval
+    - parameter completion:          The completion closure used to notify when the server
+  **/
+
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Client.swift:387]:
```suggestion
```

File: Sources/MeiliSearch/Error.swift [hunk 2]
```diff
@@ -68,6 +68,8 @@ public extension MeiliSearch {
     /// The input or output JSON is invalid.
     case invalidJSON
 
+    // TimeOut is reached in a waiting function
+    case timeOut(timeOut: Double)
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Error.swift:72]:
```suggestion
    case timeOut(timeOut: Double)

```

File: Sources/MeiliSearch/Model/Update.swift [hunk 3]
```diff
@@ -11,6 +11,9 @@ public struct Update: Codable, Equatable {
   /// The UID of the update.
   public let updateId: Int
 
+  public init(updateId: Int) {
+    self.updateId = updateId
+  }
```
Comment (bidoubiwa) [inline Sources/MeiliSearch/Model/Update.swift:16]:
```suggestion
  }

```

**Review by alallema — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**